### ISSUE TITLE
Plugin embeds all transitive dependencies/artifacts, even if they are compile dependencies

### ISSUE THREAD

**Issue by michalbrz** [Vigi0303/fat-aar-plugin#3]
I have an android library which has dependency to android module, which has dependency to pure java module. So A1 -> A2 -> J
The problem is that the plugin tries to pack into aar all dependencies from A2 and J, even if they are 'compile' dependencies (which would be usually the case). So if my J module has 'compile' dependency to RxJava, RxJava will be packed inside final aar. I assume it is not desired behavior, right? 

This is caused by embedConf.resolvedConfiguration.resolvedArtifacts, which returns every transitive artifact, but I'm not sure how to fix it and what exactly is the desired behaviour in this case.

**Comment by vigidroid** [comment]
Hi michalbrz, thanks for feeding back!

In my opinion, embedding a dependency means to embed the whole one.
Like,
```gradle
embed project1
```
project1 depends on project2(with `compile`), maybe project2 also depends on others...
project1 is a whole one that includes all others.

If you want to embed A2 excluding J, there are two ways to achieve this:
1. `provided` J in A2's `build.gradle` file. J will be excluded.
2. `embed` A2 with `@aar` classifier in A1's `build.gradle` file. All dependencies in A2 will be excluded.
e.g. 
```gradle
embed 'me.vigi:library:1.0.0@aar'
```

**Comment by michalbrz** [comment]
Thanks for your quick response.

Before I was using https://github.com/adwiv/android-fat-aar to make a library and I wanted to switch to your plugin, which looks really promising. However, android-fat-aar works like I described in this issue - it doesn't include transitive dependencies in output aar.

But after I gave it some thought, I think your approach makes sense. If `compile` doesn't embed dependency, then what is the difference between `compile` and `provided`?

So you 1st solution works nice, thanks again and keep up the great work!
